### DISCUSSION TITLE
Wait for plot to copy

### DISCUSSION THREAD

**Discussion by ghost** [guydavis/machinaris#781] [Ideas]
Is there a way to pass to plotman or madmax the -w, --waitforcopy    Wait for copy to start next plot variable?
plotman?? or madmax?? is starting new plotting before it has moved plot out of tmp dirs.  We can save some time plotting by letting it use full speed for moving the plot and plotting afterwards. My son wrote the fork for madmax and we saved good time by using it. but i do not see it happening in machinaris.

**Comment by guydavis** [comment]
Greetings!  Yes, I added pass-thru support for the `waitforcopy` parameter of Madmax into [Plotman a year ago](https://github.com/guydavis/plotman/blame/development/src/plotman/plotters/madmax.py#L28).

Here is the configuration to set on the Settings | Plotting page, note the `waitforcopy: False` under the `madmax:` section:
![image](https://user-images.githubusercontent.com/12936/194590570-0571714d-74fd-4f22-8421-18b3a6407489.png)

When plotting, you can check the `chia_plot` (madmax) command line inside the Machinaris container using `ps` or similar to see the args.  The per-job logging file in `/root/.chia/plotman/logs` (sort by newest) should also show the madmax command used.

However, Plotman launches Madmax one plot at a time, not for [n-plots](https://github.com/guydavis/machinaris/wiki/MadMax#n-plots), so `waitforcopy` doesn't really apply in most cases. 

I think most people are better served to use the **5:0** [phase schedule](https://github.com/guydavis/machinaris/wiki/MadMax) instead however:
![image](https://user-images.githubusercontent.com/12936/194590860-4317d361-7a7b-400d-97b6-7a55dae247e8.png)

Please let me know how they work for you.  Hope this helps.

**Comment by ghost** [comment]
first of you are the man :) great improvements from last year when i last touched this hehe. coming back to mining chia this time with big goals in mind y petabyte at a time.
second 
yeah i figured the wait would not work but let me try the 5 0 setting and let you know cheers.

**Comment by ghost** [comment]
ok finally after destroying 4 old hard drives i used on raid 0 for tempdir 1 and riging 3 15k rpm 2.5 inch hard drives I was able to plot in a stable manner so that I could test settings. Turns out this is the setting that does what I need
  
 How often the daemon wakes to consider starting a new plot job, in seconds.
        polling_time_s: 600
        
I set that variable at 10 minues because it give my system enough time to move the file to final destination without being stalled for too long.  The setting is all up to your system and how you like things my setting is good enough for me but you may want to go higher or lower.

Cheers

**Comment by ghost** [comment]
Update. I found the configuration that works perfect. 
        tmpdir_stagger_phase_major: 5
        tmpdir_stagger_phase_minor: 1
        # Optional: default is 1
        tmpdir_stagger_phase_limit: 0
that makes plot man wait untill it is completely done moving the finished plot. which shaved a few seconds of the whole process since harddrives are not fighting for IO because the new plot started.

 How often the daemon wakes to consider starting a new plot job, in seconds.
 polling_time_s: 20

 that pooling setting also helps but does not work as good as the above. So leave polling time 20 secs or lower but you may spam the hell out of the system hehe and use the settings above and you are good to go.
